openedx/features/genplus_features/genplus_teach/api/v1/views.py

```diff
@@ -333,3 +334,42 @@ def list(self, request, *args, **kwargs):
         alert_bar = get_object_or_404(AlertBarEntry, is_current=True)
         serializer = self.serializer_class(alert_bar)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PortfolioReflectionView(generics.ListAPIView):
+    authentication_classes = [SessionAuthenticationCrossDomainCsrf]
+    permission_classes = [IsAuthenticated, IsTeacher]
+    serializer_class = PortfolioReflectionSerializer
+
+    def get_queryset(self):
+        queryset = PortfolioReflection.objects.all()
+        teacher = Teacher.objects.get(gen_user=self.request.user.gen_user)
+
+        search = self.request.query_params.get('search', '')
+        skill = self.request.query_params.get('skill')
+        gtcs = self.request.query_params.get('standard')
+
+        Q1 = Q(portfolio__teacher=teacher)
+        Q2 = Q(reflection__teacher=teacher)
+
+        Q3 = Q(portfolio__title__icontains=search)
+        Q4 = Q(portfolio__description__icontains=search)
+        Q5 = Q(reflection__answer__icontains=search)
+        Q6 = Q(reflection__reflection__title__icontains=search)
+        
+        Q7 = Q(portfolio__skill__id=skill)
+        Q8 = Q(portfolio__gtcs__id=gtcs)
+        
+        if skill or gtcs:
+            query = Q1 & (Q3 | Q4)
+            
+            if skill:
+                query &= Q7
+            
+            if gtcs:
+                query &= Q8
+        else:
+            query = (Q1 & (Q3 | Q4)) | (Q2 & (Q5 | Q6))
+
```

Comment:
@talhaaslam01 can we change it to something like this and completely remove Q variables?
```
query = Q(portfolio__teacher=teacher) & (Q(portfolio__title__icontains=search) | Q(portfolio__description__icontains=search))

if skill:
    query &= Q(portfolio__skill__id=skill)
if gtcs:
    query &=  Q(portfolio__gtcs__id=gtcs)

if not (skill or gtcs):
    query |= (Q(reflection__teacher=teacher) & (Q(reflection__answer__icontains=search) | Q(reflection__reflection__title__icontains=search)))
```
Also it isn't clear why reflection filters relate to the absence of both skill and gtcs.

Reply:
@hasnain-saeed Reflection does not have skill and gtcs fields

Reply:
`if not (skill or gtcs)` If this conditioned is removed then, the reflections are fetched regardless of the applied filters